pkgs/applications/virtualization/lima/default.nix

```diff
@@ -48,27 +53,33 @@ buildGoModule rec {
     export LIMA_HOME="$(mktemp -d)"
   '';
 
-  installPhase = ''
-    runHook preInstall
-    mkdir -p $out
-    cp -r _output/* $out
-    wrapProgram $out/bin/limactl \
-      --prefix PATH : ${lib.makeBinPath [ qemu ]}
-  '' + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
-    installShellCompletion --cmd limactl \
-      --bash <($out/bin/limactl completion bash) \
-      --fish <($out/bin/limactl completion fish) \
-      --zsh <($out/bin/limactl completion zsh)
-  '' + ''
-    runHook postInstall
-  '';
+  installPhase =
+    ''
+      runHook preInstall
+      mkdir -p $out
+      cp -r _output/* $out
+      wrapProgram $out/bin/limactl \
+        --prefix PATH : ${lib.makeBinPath [ qemu ]}
+    ''
+    + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+      installShellCompletion --cmd limactl \
+        --bash <($out/bin/limactl completion bash) \
+        --fish <($out/bin/limactl completion fish) \
+        --zsh <($out/bin/limactl completion zsh)
+    ''
+    + ''
+      runHook postInstall
+    '';
 
   doInstallCheck = true;
   installCheckPhase = ''
     USER=nix $out/bin/limactl validate templates/default.yaml
   '';
 
-  passthru.updateScript = nix-update-script { };
+  passthru = {
+    tests.version = testers.testVersion { package = lima; };
```

Comment:
Could you use `versionCheckHook` instead?
> It is also possible to use passthru.tests to test the version with [testVersion](https://nixos.org/manual/nixpkgs/unstable/#tester-testVersion), but since that is pretty trivial and recommended thing to do, we recommend using [versionCheckHook](https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook) for that, which has the following advantages over passthru.tests:

https://nixos.org/manual/nixpkgs/unstable/#var-passthru-tests-packages

Reply:
Thank you for reviewing! I had learned versionCheckHook yesterday at https://github.com/NixOS/nixpkgs/pull/358956#discussion_r1871002548.

However, versionCheckHook ignores environment variables even if they are defined in `preVersionCheck`. So, I don't know how to resolve the [`panic: $HOME is not defined`](https://github.com/lima-vm/lima/blob/v1.0.2/pkg/limayaml/defaults.go#L52) in this package...

I've tried creating special test code with writeShellApplication, but I don't know how to inject lima's `placeholder "out"` in that. 🙇‍♂️

Reply:
How about this direction? https://github.com/NixOS/nixpkgs/pull/361378/commits/f28ecbe89ad9011c0ab45a42a6ed361db2f084e9

At least, age and several packages using these test.

https://github.com/NixOS/nixpkgs/blob/b59eb3364e8ced9912a076a26205ef870ebdbd02/pkgs/by-name/ag/age/package.nix#L27-L34

Reply:
Awesome! Thank you.